packages/strapi-admin/admin/src/containers/Users/ListPage/ModalForm.js

```diff
@@ -32,15 +32,14 @@ const ModalForm = ({ isOpen, onClosed, onToggle }) => {
   };
 
   const handleClosed = () => {
-    setStep('create');
-
     // Fetch data only if the user has submitted a new entry
     // We can use the registrationToken to know this
-    if (registrationToken) {
+    if (registrationToken || currentStep === 'magic-link') {
```

Comment:
I'm not feeling very comfortable with this notion of "magic link" since it can be almost everything 😅

Reply:
Yes, you are right. It can be confusing. The magic-link step of the modal is the equivalent of a "success modal form view"